### PR TITLE
AutoXHR: Adding xhr.ru (response URL) beacon parameter for XHR and Fetch

### PR DESCRIPTION
**Pull request by sukratkashyap**
Adding new `ru` beacon parameter which contains the Response Url when XHR request is made.

- `XMLHttpRequest`: https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/responseURL
- `fetch`: https://developer.mozilla.org/en-US/docs/Web/API/Response/url

### CODE REVIEW

File: plugins/rt.js [hunk 1]
```diff
@@ -1572,6 +1576,10 @@
 				if (edata.responseBodyNotUsed) {
 					BOOMR.addVar("fetch.bnu", 1, true);
 				}
+
+				if (edata.responseUrl) {
+					BOOMR.addVar("ru", edata.responseUrl, true);
```
Comment (bluesmoon) [inline plugins/rt.js:1581]: I'd suggest adding a namespace before the `ru`, something like `xhr.ru` to avoid clashes with other plugins.

Comment (nicjansma) [inline plugins/rt.js:1581]: Make sure to run it through `BOOMR.utils.cleanupURL(url, edata.responseUrl)` so the query string can be stripped if needed.

File: plugins/rt.js [hunk 2]
```diff
@@ -70,6 +70,10 @@
  * * `fetch.bnu`: For XHR beacons from fetch API requests, `1` if fetch response body was not used.
  * * `rt.tt`: Sum of load times across session
  * * `rt.obo`: Number of pages in session that did not have a load time
+ * * `ru`: final response URL after any redirects
+ *    - `XMLHttpRequest`: it will be present, even if no redirect happened.
```
Comment (nicjansma) [inline plugins/rt.js:74]: If there were no redirects, would this always be the same as the request URL? If they're the same, I don't think we'd want to include it on the beacon (twice).

File: tests/page-templates/07-autoxhr/52-xhr-response-url.html [hunk 3]
```diff
@@ -0,0 +1,42 @@
+<%= header %>
+<!--
+IE9 (and older) and phantomjs do not fire 'loadend' for XHRs, no autoxhr beacon was being sent
```
Comment (nicjansma) [inline tests/page-templates/07-autoxhr/52-xhr-response-url.html:3]: Can you detect those browsers easily?  If they'd "fail" this test, I'd prefer the `numBeacons` be adjusted to what they send and then `this.skip()` the tests so they still "pass".

**Comment by sukratkashyap** [comment]
@nicjansma @bluesmoon Thank you guys for the comments. I have made the changes as suggested.

File: plugins/rt.js [hunk 4]
```diff
@@ -70,6 +70,10 @@
  * * `fetch.bnu`: For XHR beacons from fetch API requests, `1` if fetch response body was not used.
  * * `rt.tt`: Sum of load times across session
  * * `rt.obo`: Number of pages in session that did not have a load time
+ * * `ru`: final response URL after any redirects
```
Comment (nicjansma) [inline plugins/rt.js:73]: Small update to `ru` -> `xhr.ru` here!

**Comment by nicjansma** [comment]
Not related to your changes, but looks like E2E tests are failing due to:

```
E/launcher - SessionNotCreatedError: session not created: This version of ChromeDriver only supports Chrome version 84
```

In our internal repo, we lock webdriver and chromedriver at specific versions during CI tests.  I'll need to see how easy that would be to do in the Github actions files.

**Comment by SaschaBrechmannVHV** [comment]
Hi, could this PR merged? The new xhr.ru variable looks good. 
Regards, Sascha

**Review by nicjansma — APPROVED** [review]

**Comment by nicjansma** [comment]
Yes!  Merged via dd558a515be0d730ca8479b2d5ef8e0dcedcd253

**Pull request closed without merging**